install-dev/data/db_structure.sql

```diff
@@ -1696,7 +1697,8 @@ CREATE TABLE IF NOT EXISTS `PREFIX_product_shop` (
   `active` tinyint(1) unsigned NOT NULL DEFAULT '0',
   `redirect_type` ENUM(
     '', '404', '410', '301-product', '302-product',
```

Comment:
I also wonder maybe instead of introducing new enum "default" we could use the default setting when this value is empty string? What happens now if there is empty string vs the "default" ?

Reply:
The current behavior (that I retained kept), is that if redirect_type ends up empty/null/false, it will use classic 404 page+response.

Reply:
Also, just realised there is '' in shop table, should I also remove it like you did here? :-) https://github.com/PrestaShop/PrestaShop/pull/22378

I will make the default value 'default' in both cases.

Reply:
in that case we wouldn't need to add additional "default" value in the RedirectType value object, but when loading the product we would have to check if the type is not set (if its the "empty string" one), then load the appropriate RedirectType based on configuration value.
That would also allow you to avoid chaning the default "404" redirect type in behats by instead setting up the default common configuration value `PS_PRODUCT_REDIRECTION_DEFAULT` to a 404.

though ofc I haven't considered this thoroughly, as I didin't dive too deep into those scenarios, so its just a suggestion to think about (maybe its a bad idea, but maybe good? :smile: idk :man_shrugging:)

Reply:
`Also, just realised there is '' in shop table, should I also remove it like you did here?`
Instead of removing it you can leave it empty and get rid of the "default" one. This way just avoiding the big BC break. (therefore the empty string would just act as a default behavior) :man_shrugging: